packages/contracts/contracts/discovery/Project.sol

```diff
@@ -38,13 +39,19 @@ contract Project is IProject {
     // has the project been approved by a Citizend manager
     bool public override(IProject) approvedByManager;
 
+    // has the project been approved by the legal team
+    bool public override(IProject) approvedByLegal;
+
+    bytes32 public constant CONTROLLER_ROLE = keccak256("CONTROLLER_ROLE");
+
     constructor(
         string memory _description,
         address _token,
         uint256 _saleSupply,
         uint256 _rate
     ) {
         controller = msg.sender;
+        _grantRole(CONTROLLER_ROLE, msg.sender);
```

Comment:
sorry, I'm failing to see what this is for?
If this is speicfically to add a role to the `Controller` instance, isn't that unnecessary, since we already know the `controller`, and there's only gonna be one?

Reply:
Was gonna ask the same thing. We can use ERC165 to ensure it's a controller, though

Reply:
I think you're confusing things @zamith . ERC165 is to ensure it's "anything that implements the controller interface"
accesscontrol, like in this case, it's to ensure it's "our specific controller, and not some other contract that implements the same interface"

it just happens that this is useless to check that here because we already know who the `controller = msg.sender` is, so this is redundant

Reply:
I'm not, that's what I meant. I was just adding to what you said.

Reply:
I think this was leftover from when I had that other function as the entry point for the approval of the project. This is not needed anymore I think